### DISCUSSION TITLE
collections

### DISCUSSION THREAD

**Discussion by ghost** [tighten/jigsaw#618] [General]
I think this might be the last question for a while.  When working with collections. Does the collections have a key that I can moved the collections outside the source folder and have its own folder.

**Comment by bakerkretzmar** [comment]
Like having a directory structure kind of like this?

```
site
⎿ source
  ⎿ posts
  ⎿ pages
⎿ episodes
⎿ etc.
```

where `episodes` is a collection? No, I don't think that's really possible right now.

**Reply by ghost** [reply]
can this be something that can be discuss to see if we can add some kind of key if other users wants it.. i know that is a big if right now since a lot of users end up using in the source folder.. 
it would be like
```
site
⎿ source
  ⎿ pages
⎿ collections
    L _posts
    L _categories
```

**Reply by bakerkretzmar** [reply]
Can you explain more about what you need that for? Why do you want them separate like that instead of putting everything in `source`?

**Reply by ghost** [reply]
so it only make sense for me to do this way if i want to keep those separate this way, if I decide to use git submodule, I can always import them to that collections folder outside of the source if i want to change my design of my website then import them.. The collections folder will never be touch whatsoever and i don't have to import them manually if my design changes or if i use different brnches for design layouts.

**Reply by bakerkretzmar** [reply]
I'm not sure I fully understand—everything you're describing there seems like it would be possible with just a regular git repo and collections in folders inside `source`. Can you explain more about why that doesn't work for you?

**Reply by ghost** [reply]
so let me explain so both of us are on the same track. What I meant was let's say I have a folder structure like this where I have a build key with `source => public` and right now all of the layouts, collections will go to the public folder. As you mention, is not possible to move the collections outside the source folder or the public folder depending on if i set the build key.
```
app
public
  _posts
  _portfolios
  _views
  404.md
  about.md
  index.md
resources
config.php
config.production.php
composer.json
helpers.php
package.json
webpack.mix.js
```
I wanted to do is move all the collections out of the public folder so that only the layouts, and blade systems are in the public folder
```
app
collections
  _posts
  _portfolios
public
  _views
  404.md
  about.md
  index.md
resources
config.php
config.production.php
composer.json
helpers.php
package.json
webpack.mix.js
```
The reason why i brought this up is becuse if I have collections setup outside of the source or public folder... I can create the collections folder as a submodule/subtree where it only imports from anotehr git repo, even if I can do that in the source folder or public, i would hve to ignore a lot of stuff which is kind of annoying at first.

I hope this make sense.

**Reply by ghost** [reply]
this way, i can create multiple branches on the same working repo and just import the collections without the need to manually add them to each branches. cuz that would be a lot of work to deal with.

**Reply by bakerkretzmar** [reply]
Ahh okay I get it, thanks. You can't really do that currently, no. You can move the views and some js/css files out of the source directory but things like `index.md` and your compiled assets do still need to live directly in the source. I agree it would be great if you could do this, and we're definitely going to make this easier to do in Jigsaw 2.

**Reply by ghost** [reply]
cool! would this be something that we can add and test if possible for jigsaw 2.

**Reply by bakerkretzmar** [reply]
Yes hopefully!

**Reply by ghost** [reply]
cann we add this to the issues or new feature...
